### PR TITLE
Disallow editing metadata of frozen worksheets; enable unfreezing worksheets

### PR DESCRIPTION
**Pull request by nelson-liu**
### Reasons for making this change

Part of #3150 . Enables unfreezing bundles. This didn't require any backend changes, just some changes to the CLI, since the `frozen` field in the DB is nullable anyway.

### CODE REVIEW

**Comment by percyliang** [comment]
The original point of freezing a worksheet is that you commit to a particular set of contents which is truly immutable.  The idea is that when you make an executable paper, you freeze it so people visiting a worksheet have some sort of guarantee (which you get for bundles).  But in practice no one uses this feature...

What's your use case?

**Comment by nelson-liu** [comment]
I actually want to use this feature :) . I also want a parallel `freeze` / `unfreeze` for bundles (that would prevent deletion). 

However, I'm nervous about accidentally freezing the worksheet (and needing to change something later) / the "permanent" aspect. I personally care less about showing people that it was permanently frozen on a date, and more that I don't accidentally delete it in the future by doing something dumb. Besides, the permanence isn't all that permanent, since you can get around it client-side----might as well expose it.

> you freeze it so people visiting a worksheet have some sort of guarantee (which you get for bundles)

There's currently no way to tell if / when a worksheet is frozen, right? (it doesn't show up on the web UI, and there's no `winfo`)

**Comment by percyliang** [comment]
Exposing the frozen bit in the UI is pretty easy to do (just never got around to it).

Freezing the worksheet I think doesn't protect the bundles on the worksheet if you do `cl rm -f`.  Can you say a bit more about the protection you have in mind?

In general, since I don't think anyone else is using this feature, you might as well repurpose it for your use case, but just want to understand it a bit better.

**Comment by nelson-liu** [comment]
> Freezing the worksheet I think doesn't protect the bundles on the worksheet if you do cl rm -f. Can you say a bit more about the protection you have in mind?

Right, I think that's fine (I recognize that `-f` is very dangerous). A lot of how i interact with codalab is via `cl search`, and I often want to bulk-delete bundles (`cl rm $(cl search <something>)`). The "leaf" experiment bundles that I might put into an executable paper (i.e., things that have no downstream dependents) have no protection from an errant `cl rm $(cl search)`. I'm also perhaps overcautious after getting bitten by this once, and I just want to ensure that I can work with reckless abandon and not worry about modifying any executable papers i have. https://github.com/codalab/codalab-worksheets/issues/2293

**Comment by nelson-liu** [comment]
I made the freeze feature even stronger---i don't see a good reason for being able to update the metadata of a frozen worksheet, especially if you can unfreeze it. Now, frozen worksheets cannot be modified at all [1], contents or metadata.

[1] There is only one attribute of a frozen worksheet that you can modify: whether or not it's frozen.

File: tests/cli/test_cli.py [hunk 1]
```diff
@@ -1077,26 +1077,47 @@ def test(ctx):
...
+    _run_command([cl, 'wedit', '--unfreeze'])
+
+    # After unfreezing: can modify everything
+    _run_command([cl, 'detach', uuid1])  # would remove an item
+    _run_command([cl, 'rm', uuid1])  # would remove an item
+    _run_command([cl, 'add', 'text', 'message'])  # would add an item
     _run_command([cl, 'wedit', '-t', 'new_title'])  # can edit
     _run_command([cl, 'wperm', wuuid, 'public', 'a'])  # can edit
 
+    # Verify that we can make multiple edits to a frozen worksheet,
```
Comment (epicfaace) [inline tests/cli/test_cli.py]: I think what this code is really verifying is that "we can make a single edit to a frozen edit, as long as we unfreeze at the same time" -- can we update the comment accordingly?

**Review by epicfaace — APPROVED** [review]

File: tests/cli/test_cli.py (hunk 1 above)
Comment (epicfaace) [reply tests/cli/test_cli.py]: @nelson-liu I just slightly updated the tests for clarity regarding this comment; let me know if you have any concerns!

Comment (nelson-liu) [reply tests/cli/test_cli.py]: Oops sorry, I must have missed this---you're correct, the new comment is more accurate. Thanks for following up!

**Pull request merged**